lib/improver/blending/blend_across_adjacent_points.py

```diff
@@ -63,19 +68,31 @@ def __init__(self, coord, width, parameter_units):
                 This does not need to be the same as the units of the
                 coordinate we are blending over, but it should be possible to
                 convert between them.
-
+            weighting_mode : string
+                The mode of blending, either weighted_mean or
+                weighted_maximum. Weighted average finds the weighted mean
+                across the dimension of interest. Maximum probability
+                multiplies the values across the dimension of interest by the
+                given weights and returns the maximum value.
+        Raises:
+            ValueError : If an invalid weighting_mode is given
         """
         self.coord = coord
         self.width = width
         self.parameter_units = parameter_units
+        if weighting_mode not in ['weighted_maximum', 'weighted_mean']:
+            msg = ("weighting_mode: {} is not recognised, must be either "
+                   "weighted_maximum or weighted_mean").format(weighting_mode)
+            raise ValueError(msg)
+        self.mode = weighting_mode
 
     def __repr__(self):
         """Represent the configured plugin instance as a string."""
-        return (
-            '<TriangularWeightedBlendAcrossAdjacentPoints:'
-            ' coord = {0:s}, width = {1:.2f},'
-            ' parameter_units = {2:s}>').format(self.coord, self.width,
-                                                self.parameter_units)
+        msg = ('<TriangularWeightedBlendAcrossAdjacentPoints:'
+               ' coord = {0:s}, width = {1:.2f},'
```

Comment:
They correspond to which argument of format goes in which {}. They aren't really needed as the default is to fill them in in order, which is fine here. I think our problems from before were because we weren't using them correctly. I'll remove them for now.